### ISSUE TITLE
Multiplying integer parameters by 16, sometimes results in the parameter being set to 0

### ISSUE THREAD

**Issue by ljg701** [boriel/zxbasic#665]
Boriel,

In the example below, the expected output would be the value 32, however a value of 0 is displayed in all cases.

If I change y*16 to y<<4 the expected output of 32 is generated. This seems to be repeatable if the previous integer parameter is also multiplied by 16. 

This is happening in the most recent beta, and also previous versions. (you'll need to go into edit mode to copy the code)

```basic
dim x,y as ubyte

for x = 0 to 10
for y = 0 to 2

printTest(x*16,y*16)

next y
next x 

do 
a$= inkey$
loop until  a$<>"" 

sub printTest(xpos as uinteger, ypos as ubyte)

print at 0,(xpos/16)*3;ypos

end sub
```

**Comment by boriel** [comment]
To enter code here, use triple quote ```

**Comment by boriel** [comment]
The 2nd example displays 16. Even changing `ypos`  parameter to `Uinteger`

**Comment by boriel** [comment]
The first example seems to work ok, too. This is the screen after the program ends:
![image](https://github.com/boriel/zxbasic/assets/1433137/6fb4fb28-206a-4192-97e5-1f0305b23f23)

This is a ZX Spectrum 48K emulator. This might be an issue with ZX Next.

**Comment by ljg701** [comment]
Thank you. Wonder if this could be caused by multiplication optimisations made for the next?

**Comment by ljg701** [comment]
Issue seems to be related to the optimiser.

A .nex file generated with the optimiser level less than 4 does not show the issue

**Comment by boriel** [comment]
Ok. Seems I could reproduce the issue!

**Comment by boriel** [comment]
Try this beta and let me know if it works for you:
http://www.boriel.com/files/zxb/zxbasic-1.17.1-beta3.tar.gz
http://www.boriel.com/files/zxb/zxbasic-1.17.1-beta3.zip
http://www.boriel.com/files/zxb/zxbasic-1.17.1-beta3-win32.zip
http://www.boriel.com/files/zxb/zxbasic-1.17.1-beta3-linux64.tar.gz
http://www.boriel.com/files/zxb/zxbasic-1.17.1-beta3-macos.tar.gz

**Comment by ljg701** [comment]
Issue seems to have been resolved with this beta. Thank you

**Comment by boriel** [comment]
Fixed in #667
